smartsim/settings/batch_settings.py

```diff
@@ -117,6 +117,15 @@ def __init__(
             """The scheduler type"""
         except ValueError:
             raise ValueError(f"Invalid scheduler type: {batch_scheduler}") from None
+
+        if batch_args is not None:
+            if not (
+                isinstance(batch_args, dict)
+                and all(isinstance(key, str) for key, val in batch_args.items())
+            ):
+                raise TypeError(
+                    "batch_args argument was not of type mapping of str and str"
+                )
```

Comment:
Minor nit: Typehint and runtime check state that this must be a `dict[str, str | None]`, but the error messages says it can be a `Mapping[str, str]`
```suggestion
        if batch_args is not None:
            if not (
                isinstance(batch_args, dict)
                and all(isinstance(key, str) for key, val in batch_args.items())
            ):
                raise TypeError(
                    "batch_args argument was not of type dict of str and str or None"
                )
```

Reply:
Also, since we are already iterating over `batch_args`, should we runtime type check the values as well as the keys of `batch_args`?

Reply:
the type hint for it is t.Dict[str, t.Optional[str]], so what should the value type check look like? how do I check for optional str?